compiler/llvm/clangUtil.cpp

```diff
@@ -4775,6 +4775,18 @@ static void makeBinaryLLVMForCUDA(const std::string& artifactFilename,
   // functions are called from the kernel
   ptxasFlags += " --suppress-stack-size-warning ";
 
+
+  // With CUDA 12.4, PIC is on by default in ptxas. However, with some large
+  // codes, we see that ptxas disables PIC and issues a warning about it. Most
+  // likely, we are not going to benefit from PIC anytime soon (I don't even
+  // know what it means for us to benefit from that? Generated GPU code as
+  // shared library... for multiple Chapel applications?). So, disable PIC for
+  // the time being.
+  const auto sdkVer = envMap["CHPL_GPU_SDK_VERSION"];
+  if (strncmp(sdkVer, "12", 2) == 0) {
+    ptxasFlags += " --position-independent-code=false ";
```

Comment:
Someone else should review this from a GPU perspective, but you might want to add a check that `CHPL_LIB_PIC` is not set to `pic` and intentionally error out if it is.  This will probably get in the way of compiling GPU code into Python libraries with NVIDIA, but I suspect that's okay for now

Reply:
How about we generate a warning for that case?

Just musing: Currently, the GPU binary is embedded into the binary we generate. From the compiler/runtime standpoint it literally is a global `const char*` of a binary blob. That blob is loaded by the runtime during start up, and the CPU launching GPU code can only happen through the runtime, i.e. the user can't write a Chapel proc that'll launch on the GPU. Given that landscape, I think the GPU binary and whether it uses PIC will be irrelevant as its use must come from the Chapel runtime/application which could still use PIC on the host. With that _very_ vague understanding of what PIC is, things will probably work, I think. Maybe. Probably?

Reply:
Seems worth a shot at least

Reply:
I shied away from making any change here, Lydia. I want to merge this today to be able to handle potential fallout tomorrow. I created https://github.com/Cray/chapel-private/issues/6942 to understand the story here a bit better.

Reply:
I think that's reasonable, it's moving things forward.  Thanks for opening the issue!